File: Problemas/1791C - Prepend and Append/1791C_Prepend_And_Append.cpp
// 1791C - Prepend and Append 
// https://codeforces.com/problemset/problem/1791/C

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas = Número de casos de prueba
    int pruebas;
    cin >> pruebas;

    // Itera sobre cada caso de prueba
    for(int indice_prueba = 0; indice_prueba < pruebas; indice_prueba++) {

        // Input:
        // 2) N = Tamaño del string
        int N;
        cin >> N;

        // Input:
        // 3) binario = String de 0s y 1s
        string binario;
        cin >> binario;

        // Variables apuntadoras
        int izquierda = 0;
        int derecha = N - 1;

        // Si las dos variables tienen vadlores distintos, entonces se recorren en direcciones opuestas
        while(binario[izquierda] != binario[derecha] && derecha > izquierda) {
            izquierda++;
            derecha--;
        }

        // Si el string original es vacío, entonces imprime 0
        if(derecha < izquierda) {
            cout << 0 << endl;
        } else {
            int longitud_original = derecha - izquierda + 1;
            cout << longitud_original << endl;
        }
    }

    return 0;
}

File: Problemas/1791C - Prepend and Append/1791C_Prepend_And_Append.py
# 1791C - Prepend and Append
# https://codeforces.com/problemset/problem/1791/C

# Input:
# 1) pruebas = Número de casos de prueba
pruebas = int(input())

# Itera sobre cada caso de prueba
for indice_prueba in range(0,pruebas):

    # Input:
    # 2) N = Tamaño del string
    N = int(input())

    # Input
    # binario = String de 0s y 1s
    binario = input()
    
    # Variables apuntadoras
    izquierda = 0
    derecha = N - 1

    # Si las dos variables tienen vadlores distintos, entonces se recorren en direcciones opuestas
    while binario[izquierda] != binario[derecha] and derecha > izquierda:
        izquierda += 1
        derecha -= 1

    # Si el string original es vacío, entonces imprime 0
    if derecha < izquierda:
        print(0)

    # Si el sring original no está vacío, calcula la distancia entre las variables apuntadoras + 1
    else:
        longitud_original = derecha - izquierda + 1
        print(longitud_original)

